internal/db/schema/migrations/postgres/14/01_event.up.sql

```diff
@@ -0,0 +1,250 @@
+│public_id        │     │   │                     
+│config_id        │     │   │                     
+│                 │     │   │                     
+└─────────────────┘     │   │                     
+         ┼ ┼            │   │                     
+         │ │            │   │                     
+         │ │            │   │                     
+         │ │            │   │                     
+         │ │            │   │                     
+         │ │            │   └────────────────┐    
+         │ │            │                    │    
+         │ └────────────┼────────┐           │    
+         ○              │        │           │    
+        ╱│╲             │        │           │    
+┌─────────────────┐     │        │           │    
+│ event_file_sink │     │        │           │    
+├─────────────────┤     │        │           │    
+│ public_id       │┼────┘        ○           │    
+│ sink_id         │             ╱│╲          │    
+│ event_type      │     ┌─────────────────┐  │    
+│ format_type     │     │event_stderr_sink│  │    
+│ allow_filters   │     ├─────────────────┤  │    
+│ deny_filters    │     │ public_id       │  │    
+│ path            │     │ sink_id         │  │    
+│ file_name       │     │ event_type      │┼─┘    
+│ rotate_bytes    │     │ format_type     │       
+│ rotate_duration │     │ allow_filters   │       
+│ rotate_max_files│     │ deny_filters    │       
+└─────────────────┘     └─────────────────┘       
+         ┼                        ┼               
+         │                        │               
+         │                        │               
+         │  ┌─────────────────┐   │               
+         │ ╱│event_format_type│╲  │               
+         └──├─────────────────┤───┘               
+           ╲│                 │╱                  
+            └─────────────────┘                                                                                     
+*/
+
+create table event_type_enm (
+    name text primary key
+        constraint only_predefined_event_types_allowed
+        check (
+            name in (
+                'every',
+                'error',
+                'audit',
+                'observation',
+                'system'
+            )
+        )
+);
+comment on table event_type_enm is
+'event_type_enm is an enumeration table for the valid event types within '
+'the domain';
+
+create table event_format_type_enm (
+    name text primary key
+        constraint only_predefined_event_format_types_allowed
+        check (
+            name in (
+                'cloudevents-json',
+                'cloudevents-text',
+                'hclog-text',
+                'hclog-json'
+            )
+        )
+);
+comment on table event_format_type_enm is
+'event_format_type_enm is an enumeration table for the valid event format types '
+'within the domain';
+
+create table event_config (
```

Comment:
Are any of the columns in the table immutable?  This question applies to other tables like: event_type_enabled, event_file_sink, event_sink, etc.